### PR TITLE
test(js): Remove warnOnMissingMocks

### PR DESCRIPTION
**Pull request by evanpurkhiser**
This adds many missing API request mocks.

Since mosts of the tests weren't actually relying on the data being there, most return empty results.

I did have to add some `await act(tick)`s for a few scenarios where the API promises were resolving later and causing state updates.

**This makes our test output _almost_ completely clean again**

### CODE REVIEW

File: static/app/actionCreators/group.spec.jsx [hunk 1]
```diff
@@ -81,12 +81,15 @@ describe('group', () => {
 
   describe('bulkUpdate()', function () {
     beforeEach(function () {
-      jest.spyOn(api, 'request');
```
Comment (evanpurkhiser) [inline static/app/actionCreators/group.spec.jsx:84]: This was doing what addMockResponse is supposed to help you do haha

File: static/app/components/globalSdkUpdateAlert.spec.tsx [hunk 2]
```diff
@@ -225,22 +225,26 @@ describe('GlobalSDKUpdateAlert', () => {
       snoozed_ts: undefined,
     };
 
-    const promptsActivityMock = MockApiClient.addMockResponse({
+    MockApiClient.addMockResponse({
       url: '/prompts-activity/',
       body: {data: promptResponse},
     });
 
+    const promptsActivityMock = MockApiClient.addMockResponse({
```
Comment (evanpurkhiser) [inline static/app/components/globalSdkUpdateAlert.spec.tsx:233]: This was never actually testing that the PUT request was happening

File: static/app/components/teamSelector.spec.jsx [hunk 3]
```diff
@@ -135,5 +138,8 @@ describe('Team Selector', function () {
       expect.objectContaining({value: '2'}),
       expect.anything()
     );
+
+    // Wait for store to be updated from API response
+    await act(tick);
```
Comment (priscilawebdev) [inline static/app/components/teamSelector.spec.jsx:143]: nooo 😢

Comment (priscilawebdev) [reply static/app/components/teamSelector.spec.jsx:143]: is not possible to wait for the element to be rendered in the UI by using some of the `findBy` or `waitFor` queries?

Comment (priscilawebdev) [reply static/app/components/teamSelector.spec.jsx:143]: for all tests i've written so far i found a way to not use `tick`

Comment (evanpurkhiser) [reply static/app/components/teamSelector.spec.jsx:143]: I dunno. I didn't want to spend a bunch of time trying to figure it out.

Comment (evanpurkhiser) [reply static/app/components/teamSelector.spec.jsx:143]: Some of these are things like the render-prop style `Projects` util, that internally update state when the store changes, but do not actually render anything.

Unfortunately that still triggers the act warnings.

**Review by scttcper — APPROVED** [review]

File: static/app/views/eventsV2/resultsChart.spec.tsx [hunk 4]
```diff
@@ -130,5 +136,8 @@ describe('EventsV2 > ResultsChart', function () {
     expect(yAxisOptions.length).toEqual(2);
     expect(yAxisOptions[0].value).toEqual('count()');
     expect(yAxisOptions[1].value).toEqual('count_unique(user)');
+
+    // Wait for event geo request results to preopgate to update
+    await act(tick);
```
Comment (ryan953) [inline static/app/views/eventsV2/resultsChart.spec.tsx:141]: at the end of the test? seems weird idk.

Comment (evanpurkhiser) [reply static/app/views/eventsV2/resultsChart.spec.tsx:141]: yeah it's weird, idk either.

If we make a concerted effort to get rid of act and tick we can revisit then.

Comment (priscilawebdev) [reply static/app/views/eventsV2/resultsChart.spec.tsx:141]: I am down for this effort 😄

**Pull request merged**